coursebuilder/tests/unit/common_utils.py

```diff
@@ -349,3 +349,40 @@ def test_urls(self):
 
         self.assertIsNone(utils.find_youtube_video_id(''))
         self.assertIsNone(utils.find_youtube_video_id(VIDEO_ID[1:]))
+
+
+class ValidateCarouselEmbedder(unittest.TestCase):
+    def test_replaces_keywords_with_carousel_components(self):
+        content = '[start_carousel] 1 [split] 2 [stop_carousel]'
+
+        self.assertEqual(
+            """<div class="owl-carousel owl-theme course-carousel"><div class="item"><p>1 </p></div><div class="item"><p>2 </p></div></div>""",
+            utils.embed_carousel(content))
+
+    def test_embeds_carousel_when_keywords_are_within_html_tags(self):
+        content = '[start_carousel] <div style="color: #fff;"><h2>1 [split]</h2></div> <p>2</p> [stop_carousel]'
+
+        self.assertEqual(
+            """<div class="owl-carousel owl-theme course-carousel"><div class="item"><div style="color: #fff;"><h2>1 </h2></div></div><div class="item"><p>2</p> </div></div>""",
+            utils.embed_carousel(content))
+
+    def test_embeds_multiple_carousels(self):
+        content = '[start_carousel] 1 [split] 2 [stop_carousel] [start_carousel] 3 [split] 4 [stop_carousel]'
+
+        self.assertEqual(
+            """<div class="owl-carousel owl-theme course-carousel"><div class="item"><p>1 </p></div><div class="item"><p>2 </p></div></div> <div class="owl-carousel owl-theme course-carousel"><div class="item"><p>3 </p></div><div class="item"><p>4 </p></div></div>""",
+            utils.embed_carousel(content))
+
+    def test_return_original_content_if_no_closing_keyword_was_added(self):
+        content = '[start_carousel] 1 [split] 2'
```

Comment:
@andela-angene Just as with validating html, we can help the user recover from some of the obvious/expected scenarios such as forgetting to close. We just need to let them know what twe did during preview.

Reply:
An error message will now be displayed to the user if he/she makes some obvious mistake, I also added the preview feature for the carousel.